libethcore/Transaction.cpp

```diff
@@ -64,10 +64,15 @@ TransactionBase::TransactionBase(bytesConstRef _rlpData, CheckTransaction _check
 			BOOST_THROW_EXCEPTION(InvalidTransactionFormat() << errinfo_comment("transaction data RLP must be an array"));
 
 		m_data = rlp[field = 5].toBytes();
-		byte v = rlp[field = 6].toInt<byte>() - 27;
+
+		byte v = rlp[field = 6].toInt<byte>();
 		h256 r = rlp[field = 7].toInt<u256>();
 		h256 s = rlp[field = 8].toInt<u256>();
 
+		m_chainId = std::floor((float)(v - 35)/2);
+		if(m_chainId == -4 || m_chainId == 1)
+			v = rlp[field = 6].toInt<byte>() - m_chainId*2 - 35;
```

Comment:
Intuitively, I think it is fine, but better to spell it like `- (m_chainId * 2 + 35)`. Please also keep spaces between binary arithmetic operators.